app/assets/javascripts/live_search.js

```diff
@@ -146,6 +146,17 @@
       var consentCookie = GOVUK.getConsentCookie()
 
       if (consentCookie && consentCookie.settings) {
+        if (this.$resultsWrapper) {
+          this.$resultsWrapper.setAttribute('data-ga4-search-query', this.currentKeywords())
+          var sortedBy = this.$resultsWrapper.querySelector('.js-order-results')
+          if (sortedBy) {
+            this.$resultsWrapper.setAttribute('data-ga4-ecommerce-variant', sortedBy.options[sortedBy.selectedIndex].text)
+          }
+        }
+        if (this.$suggestionsBlock) {
+          this.$suggestionsBlock.setAttribute('data-ga4-search-query', this.currentKeywords())
```

Comment:
Interested to understand why we set `data-ga4-search-query` with the same value on both the suggestions block and the results wrapper. I think the results wrapper is always present, but the suggestions block might/might not be?

Reply:
@andysellick Good spot. I think it's because Universal Analytics ecommerce tracking is on the suggestions block <div>, and the results block <div>. In order for the analysts to know what search term is related to the suggestions <div> and the results <div>, they both have a `data-search-query` value. So they'll always be the same because it's just the keyword used for the search, but they're both different instances of the UA ecommerce tracker, so they each need their own data attribute.

I realised that we don't even use GA4 ecommerce tracking on the suggestions block. We use link tracking for the suggestions instead, and there's also no `data-ga4-ecommerce` attribute on the suggestions. So I'll remove these lines as they're not needed.